### ISSUE TITLE
Issue with @chardelete command

### ISSUE THREAD

**Issue by stephenvogler** [evennia/evennia#1219]
#### Brief summary of issue / Description of requested feature:
An untrapped error occurs when using the @chardelete command. When this happens, you must reload Evennia from the command line.

#### Steps to reproduce the issue / Reasons for adding feature:

1.  @charcreate testdude
2.  @chardelete testdude
3.  Freak out!

#### Error output / Expected result of feature

Traceback (most recent call last):
  File "/home/svogler/mud/evennia/evennia/commands/cmdhandler.py", line 496, in _run_command
    ret = yield cmd.func()
  File "/home/svogler/mud/evennia/evennia/utils/evmenu.py", line 891, in func
    callback = caller.ndb._getinput._callback
AttributeError: 'NoneType' object has no attribute '_callback'

An untrapped error occurred.
(Traceback was logged 2017-02-16 23:00:46-0000).
Traceback (most recent call last):
  File "/home/svogler/mud/evennia/evennia/commands/cmdhandler.py", line 626, in cmdhandler
    ret = yield _run_command(syscmd, syscmd.key, sysarg, raw_string)
ErrorReported: l

A command handler bug occurred. If this is not due to a local change,
please file a bug report with the Evennia project, including the
traceback and steps to reproduce.
(Traceback was logged 2017-02-16 23:00:46-0000).

#### Extra information, such as Evennia revision/repo/branch, operating system and ideas for how to solve / implement:

Running Evennia 0.6.0 (rev 1b3fb8f) on Gentoo, Debian 8, and Ubuntu 16.10.1 all with the same result.

**Comment by stephenvogler** [comment]
As per the traceback, the issue was in evmenu.py. I modified my local copy starting at line 891 to the following and it resolved the issue:

...
        try:
            callback = caller.ndb._getinput._callback
        except:
            caller = self.player
            callback = caller.ndb._getinput._callback
        if not callback:
            raise RuntimeError("No input callback found.")
...

**Comment by Griatch** [comment]
@stephenvogler 

Your fix only handles a symptom, not the cause. I cannot reproduce this; the getinput callback works fine. which MULTISESSION_MODE are you using? Have you done any modifications to the login mechanic?

**Comment by stephenvogler** [comment]
Yeah, I knew I wasn't addressing the cause so I just wanted to paste in what I had done.

I recently switched to MULTISESSION_MODE = 3
No changes to the actual login mechanics, just pre and post puppet and unpuppet.

**Comment by Griatch** [comment]
Still cannot replicate this in MULTISESSION_MODE=3. Can you see this also on a completely vanilla Evennia?

**Comment by stephenvogler** [comment]
I did a fresh pull and inited a new game, same issue... If you can't reproduce, I'm likely missing something here. I'm going to do a 100% fresh vanilla install and find out what breaks it :)

**Comment by stephenvogler** [comment]
Ok, 100% fresh install on another drive: exact same results with oob settings (no changes to settings.py whatsover).

**Comment by Griatch** [comment]
Ok, maybe it's something in my test setup then, will try again.

**Comment by Griatch** [comment]
@stephenvogler 

I have now done multiple attempts to reproduce this. I used Linux. I went through this process: 

- I cloned a fresh evennia, set up a fresh virtualenv and made a new game folder + database
- Changed to `MULTISESSION_MODE = 3`
- Logged in, did `@charcreate testdude` and then `@chardelete testdude` I get a question if I want to delete it, select yes - works fine.
- Logged in as a non-superuser, did the same thing. Worked fine. 

Switching to `MULTISESSION_MODE = 0`, (default, so not 3 as reported) with a fresh setup *does* show the error however, so I can confirm it does exist.
